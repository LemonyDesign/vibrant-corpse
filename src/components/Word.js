import React from 'react';
import '../styles/components/word.scss';
import PropTypes from 'prop-types';

function Word({ modifier, baseword }) {
  return (
    <li>
      <span className="modifier__items" modifiers={modifier}>
        {modifier}
      </span>
      <span className="baseword__items" basewords={baseword}>
        {baseword}
      </span>
    </li>
  );
}

Word.propTypes = {
  modifier: PropTypes.string.isRequired,
  baseword: PropTypes.string.isRequired,
};

export default Word;
