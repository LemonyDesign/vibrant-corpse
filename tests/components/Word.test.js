import React from "react";
import renderer from "react-test-renderer";
import Word from "../../src/components/Word";

describe("Word", () => {
  test("should render correctly", () => {
    const tree = renderer
      .create(<Word modifier="Crazy" photo="Monkey" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
