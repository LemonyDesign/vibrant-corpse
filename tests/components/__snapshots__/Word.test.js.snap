// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Word should render correctly 1`] = `
<li>
  <span
    className="modifier__items"
    modifiers="Crazy"
  >
    Crazy
  </span>
  <span
    className="baseword__items"
  />
</li>
`;
